x-pack/platform/plugins/shared/lens/public/app_plugin/shared/edit_on_the_fly/types.ts

```diff
@@ -86,6 +86,8 @@ export interface EditConfigPanelProps {
   // in cases where the embeddable is not filtered by time
   // (e.g. through unified search) set this property to true
   hideTimeFilterInfo?: boolean;
+  /** The dashboard api, important for creating controls from the ES|QL editor */
+  dashboardApi?: DashboardApi;
```

Comment:
I think this should be more generice and be something like `parentApi: unknown`. Then, type guards should be used to see if parentApi exposes items like `children$`, `controlGroupApi$`, and `esqlVariables$`. Lens embeddables are used in lots of places that are not dashboard and these locations should be able to support ES|QL controls without needing to implement the entire `DashboardApi`

Reply:
Done Nathan https://github.com/elastic/kibana/pull/202875/commits/5c8248c91694698a674b48a234f9b482ebec85d6